Sources/MapboxNavigation/NavigationView.swift

```diff
@@ -229,34 +236,194 @@ extension NavigationView {
+        let compactTopBannerContainerViewTrailingConstraint = topBannerContainerView.trailingAnchor.constraint(equalTo: trailingAnchor,
+                                                                                                               constant: -(screenWidth * 0.6))
+        let compactTopBannerContainerViewTopConstraint = topBannerContainerView.topAnchor.constraint(equalTo: topAnchor)
+        
+        let compactTopBannerConstraints = [
+            compactTopBannerContainerViewLeadingConstraint,
+            compactTopBannerContainerViewTrailingConstraint,
+            compactTopBannerContainerViewTopConstraint
+        ]
+        
+        let regularFloatingStackViewTopConstraint = floatingStackView.topAnchor.constraint(equalTo: topBannerContainerView.bottomAnchor, constant: 10)
+        let compactFloatingStackViewTopConstraint = floatingStackView.topAnchor.constraint(equalTo: safeTopAnchor, constant: 10)
+        
+        switch traitCollection.verticalSizeClass {
+        case .unspecified:
+            fallthrough
+        case .regular:
+            NSLayoutConstraint.activate(regularTopBannerConstraints)
+            NSLayoutConstraint.deactivate(compactTopBannerConstraints)
+            
+            NSLayoutConstraint.activate([regularFloatingStackViewTopConstraint])
+            NSLayoutConstraint.deactivate([compactFloatingStackViewTopConstraint])
+        case .compact:
+            NSLayoutConstraint.activate(compactTopBannerConstraints)
+            NSLayoutConstraint.deactivate(regularTopBannerConstraints)
+            
+            NSLayoutConstraint.activate([compactFloatingStackViewTopConstraint])
+            NSLayoutConstraint.deactivate([regularFloatingStackViewTopConstraint])
+        @unknown default:
+            break
+        }
+
+        let regularBottomBannerContainerViewTrailingConstraint = bottomBannerContainerView.trailingAnchor.constraint(equalTo: trailingAnchor)
+        let regularBottomBannerContainerViewLeadingConstraint = bottomBannerContainerView.leadingAnchor.constraint(equalTo: leadingAnchor)
+        let regularBottomBannerContainerViewBottomConstraint = bottomBannerContainerView.bottomAnchor.constraint(equalTo: bottomAnchor)
 
-        floatingStackView.topAnchor.constraint(equalTo: topBannerContainerView.bottomAnchor, constant: 10).isActive = true
+        let regularBottomBannerConstraints = [
+            regularBottomBannerContainerViewTrailingConstraint,
+            regularBottomBannerContainerViewLeadingConstraint,
+            regularBottomBannerContainerViewBottomConstraint
+        ]
+        
+        let compactBottomBannerContainerViewTrailingConstraint = bottomBannerContainerView.trailingAnchor.constraint(equalTo: trailingAnchor,
+                                                                                                                     constant: -(screenWidth * 0.6))
```

Comment:
In the iPhone SE screenshot you provided, the distance and time labels collide into each other. CarPlay removes the possibility of collisions by wrapping units onto a second line, but that would be a more complex change. Perhaps we could hide the distance label when there isn’t enough room.

It’s a little unusual that the ✖️ close button hangs out on the right end, somewhere in the middle of the screen. Moving it out of the bottom banner would help make things fit better. Would it be straightforward to replace it with a floating button in the bottom-right corner of the map? That way the close button is in a predictable location on screen, and the ℹ️ attribution button doesn’t seem like it’s floating out in the middle of nowhere.

Reply:
Regarding close button: it's a bit tricky to movie it at this point without breaking several APIs. I can also see that most of the navigation apps on the market also show close button in bottom bar menu. I've changed position of attribution button to take into account orientation and notch presence.

I'll play around with a possibility of hiding secondary label in bottom menu, this is indeed an issue on devices with smaller screen sizes.